processing/src/test/java/org/apache/druid/query/groupby/NestedQueryPushDownTest.java

```diff
@@ -0,0 +1,1059 @@
+  {
+    QueryToolChest<Row, GroupByQuery> toolChest = groupByFactory.getToolchest();
+    GroupByQuery pushDownQuery = nestedQuery;
+    QueryRunner<Row> segment1Runner = new FinalizeResultsQueryRunner<Row>(
+        toolChest.mergeResults(
+            groupByFactory.mergeRunners(executorService, getQueryRunnerForSegment1())
+        ),
+        (QueryToolChest) toolChest
+    );
+
+    QueryRunner<Row> segment2Runner = new FinalizeResultsQueryRunner<Row>(
+        toolChest.mergeResults(
+            groupByFactory2.mergeRunners(executorService, getQueryRunnerForSegment2())
+        ),
+        (QueryToolChest) toolChest
+    );
+
+    QueryRunner<Row> queryRunnerForSegments = new FinalizeResultsQueryRunner<>(
+        toolChest.mergeResults(
+            new QueryRunner<Row>()
+            {
+              @Override
+              public Sequence<Row> run(QueryPlus<Row> queryPlus, Map<String, Object> responseContext)
+              {
+                return Sequences
+                    .simple(
+                        ImmutableList.of(
+                            Sequences.map(
+                                segment1Runner.run(queryPlus, responseContext),
+                                toolChest.makePreComputeManipulatorFn(
+                                    (GroupByQuery) queryPlus.getQuery(),
+                                    MetricManipulatorFns.deserializing()
+                                )
+                            ),
+                            Sequences.map(
+                                segment2Runner.run(queryPlus, responseContext),
+                                toolChest.makePreComputeManipulatorFn(
+                                    (GroupByQuery) queryPlus.getQuery(),
+                                    MetricManipulatorFns.deserializing()
+                                )
+                            )
+                        )
+                    )
+                    .flatMerge(Function.identity(), queryPlus.getQuery().getResultOrdering());
+              }
+            }
+        ),
+        (QueryToolChest) toolChest
+    );
+    GroupByStrategy strategy = ((GroupByQueryRunnerFactory) groupByFactory).getStrategySelector()
+                                                                           .strategize(nestedQuery);
+    // Historicals execute the query with force push down flag as false
+    GroupByQuery queryWithPushDownDisabled = pushDownQuery.withOverriddenContext(ImmutableMap.of(
+        GroupByQueryConfig.CTX_KEY_FORCE_PUSH_DOWN_NESTED_QUERY,
+        false
+    ));
+    Sequence<Row> pushDownQueryResults = strategy.mergeResults(
+        queryRunnerForSegments,
+        queryWithPushDownDisabled,
+        context
+    );
+    GroupByQuery rewrittenQuery = ((GroupByQueryQueryToolChest) toolChest).rewriteNestedQueryForPushDown(nestedQuery);
+    // Broker executes this code and hence has
+    return strategy.applyPostProcessing(strategy.processSubqueryResult(
```

Comment:
Can we call `GroupByQueryQueryToolChest.mergeResults()` instead of calling this directly? I think it would be better because the unit test would execute the same logic with the production code.